DOF:
  joint_names: ['base_roll_joint', 'shoulder_lift_joint', 'shoulder_roll_joint', 'elbow_lift_joint', 'elbow_roll_joint', 'wrist_lift_joint', 'wrist_roll_joint']
  motor_names: ['base_roll_motor', 'right_motor1', 'left_motor1', 'right_motor2', 'left_motor2', 'right_motor3', 'left_motor3']
  gear_ratio: [7.1875, 7.1875, 8.24448529412, 7.1875, 8.24448529412, 7.1875, 8.24448529412] # 8.24448529412 results from 7.1875 * (39 / 34) from the output gear
  current_slope: [0.626, 0.626, 0.626, 0.626, 0.626, 1.0, 1.0] # 1.0 is decent approx for small motors
  current_offset: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
  calibration_initial_angles: [-0.62, 0.99, 0.0, -0.83, 0.0, 0.0, 0.0]
  hardstop_start_angles: [0.0, -0.17, 0.0, 0.0, 0.0, 0.0, 0.0] # angles that you the robot is in for hardstop calibration
  directions:        [1.0, -1.0, 1.0, -1.0, 1.0, -1.0, 1.0]
  torque_directions: [1.0, -1.0, 1.0, -1.0, 1.0, -1.0, 1.0]
  calibration_encoder: [4.4834423477927574, 4.4834423477927574, 3.5856800916826863, 4.4834423477927574, 3.5856800916826863, 4.4834423477927574, 3.5856800916826863] # deprecated
  calibration_joint_vive: [-0.2689415593931021, -0.2689415593931021, 0.085327798760949689, -0.2689415593931021, 0.085327798760949689, -0.2689415593931021, 0.085327798760949689] #deprecated
  #vive or rviz
  target_mode: "vive"
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 500  
  hardstop_torque_limit: 0.1
  forearm_tracker: 'forearm_tracker_link'
  ee_tracker: 'ee_tracker_link'
  # Position Controllers ---------------------------------------
  simple_controller:
    endlink: 'wrist_roll_link'
    paired_constraints: [1, 2, 3, 4, 5, 6]
    filter_length: 20
    type: koko_controllers/InverseDynamicsController
    root_name: DOF
    base_roll_joint:
      p: 100
      d: 0.0
      id: 1.
      max_torque: 30
      min_torque: -30
      max_angle: 1.6
      min_angle: -1.4  
    shoulder_lift_joint:
      p: 100
      d: 0.0
      id: 1
      max_torque: 30
      min_torque: -30
      max_angle: 3.14
      min_angle: -1.4  
    shoulder_roll_joint:
      p: 100
      d: 0.0
      id: 1.0
      max_torque: 30
      min_torque: -30
      max_angle: 1.5
      min_angle: -1.5  
    elbow_lift_joint:
      p: 80
      d: 0.0
      id: 1
      max_torque: 30
      min_torque: -30
      max_angle: 3.5
      min_angle: -1.4  
    elbow_roll_joint:
      p: 50
      d: 0.0
      id: 1.0
      max_torque: 30
      min_torque: -30
      max_angle: 1.5
      min_angle: -1.5
    wrist_lift_joint:
      p: 50.0
      d: 0.0
      id: 0.0 
      max_torque: 15
      min_torque: -15
      max_angle: 3.4
      min_angle: -3.4  
    wrist_roll_joint:
      p: 30.0
      d: 0.0
      id: 0.0
      max_torque: 15
      min_torque: -15
      max_angle: 3.4
      min_angle: -3.4



